server/mongoDB.ts

```diff
@@ -38,27 +38,29 @@ if (!g.mongo) {
   g.mongo = {};
 }
 
-export async function connectToDatabase(): Promise<MongoCtx> {
+export async function connectToDatabase(
+  collectionName: string
+): Promise<MongoCtx> {
   // we know because of L36 that this is defined
   const cached = g.mongo!;
 
-  // if there already is a connection, then use it
-  if (cached.conn) {
+  // if cached is to correct collection, use it
+  if (
+    cached.conn &&
+    collectionName === cached.conn.dataCollection.collectionName
+  ) {
     return cached.conn;
   }
 
   // instantiate to a promise resolved with the context
-  // do this immediately, and only do this once
-  if (!cached.promise) {
-    cached.promise = MongoClient.connect(MONGODB_URI, {
-      useNewUrlParser: true,
-      useUnifiedTopology: true,
-    }).then((client) => {
-      const db = client.db(MONGODB_DB);
-      const applicantDataCollection = db.collection('applicant_data');
-      return { client, db, applicantDataCollection };
-    });
-  }
+  cached.promise = MongoClient.connect(MONGODB_URI, {
```

Comment:
I may be wrong, but if we have a cached.conn, we must have a promise for the same collection right? Since a MongoCTX object has both?